artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/converter/message/JMSMappingOutboundTransformerTest.java

```diff
@@ -361,7 +361,7 @@ public void testConvertEmptyObjectMessageToAmqpMessageWithAmqpValueBody() throws
       assertNotNull(amqp.getBody());
       assertTrue(amqp.getBody() instanceof AmqpValue);
       assertTrue(((AmqpValue) amqp.getBody()).getValue() instanceof Binary);
-      assertEquals(0, ((Binary) ((AmqpValue) amqp.getBody()).getValue()).getLength());
+      assertEquals(5, ((Binary) ((AmqpValue) amqp.getBody()).getValue()).getLength());
```

Comment:
At first glance this looks wrong as I'd have expected the sender to have sent a zero length Binary and so the resulting message should also contain a zero length Binary as the value in the encoded AmqpValue.

Reply:
I thought the same, but I looked around at the other tests and several of them were using 5 in their assertions.  If the assertion uses 0 the test fails so that indicates a problem with the implementation rather than the test, of course.

Reply:
I'm back to work on Monday and can look closer, a quite look at the equivalent tests in 5.x show that they all test for 0 as the expected length so I'm not sure why these are checking for 5 now, that seems wrong to me but I'd have to look closer when I'm back.

Reply:
@tabish121 what about reverting the change on the AMQP converter and just avoid the conversion? This will only be relevant if converting back to AMQP, so not converting to Core just because of large messages would be the best choice.

Reply:
Would it not be valid to support cross protocol large message?

Fair enough for no cross protocol if you have a more elegant solution. But having cross protocol also is good

Reply:
@clebertsuconic once you have something better that is tested and working, sure you can revert this, but for now those of us getting messages that fall into the "large" message trap see unreliable transformations so this fix is needed.  Once the broker does it better then you can removed these workarounds if you know for sure that conversions are only every one way.

Reply:
@michaelandrepearce I dont beleive Clebert was talking about removing cross-protocol support, but simply noting that the issue I originally reported only occurred because the broker converted the AMQP message to a 'large' Core message and back (with a loss of detail prior to Tim's changes), when in fact no protocol conversion was really needed since only AMQP producers and consumers were present.

Reply:
@gemmellr that is correct!

@tabish121 the fix you added only works if AMQP->Core->AMQP.. it has nothing to do with the conversion itself. So.. I will keep the work around for now.. but I consider fixing it quite a blocker for releasing master again. I will add a JIRA with a blocker uppercase.. meaning.. we need to fix LargeMesasges for AMQP.

Reply:
i'm adding https://issues.apache.org/jira/browse/ARTEMIS-1975

Reply:
Reviewing the changes and tests and all looks good, the unexpected checks for content length in the object message bodies seems to be a side effect of the way the test constructs the message and not related to any new bugs.

Reply:
Thanks, Tim.